ddtrace/logger.py

```diff
@@ -0,0 +1,40 @@
+import logging
+from logging.handlers import RotatingFileHandler
+import os
+
+from ddtrace.internal.utils.formats import asbool
+
+
+DEFAULT_FILE_SIZE_BYTES = 15 << 20  # 15 MB
+
+
+def configure_ddtrace_logger():
+    # type: () -> None
+    """ """
+    debug_enabled = asbool(os.environ.get("DD_TRACE_DEBUG", "false"))
+    log_path = os.environ.get("DD_TRACE_LOG_FILE")
+    max_file_bytes = int(os.environ.get("DD_TRACE_FILE_SIZE_BYTES", DEFAULT_FILE_SIZE_BYTES))
+    num_backup = 1
+    log_format = "%(asctime)s %(levelname)s [%(name)s] [%(filename)s:%(lineno)d] {}- %(message)s"
+    log_formatter = logging.Formatter(log_format)
+
+    ddtrace_logger = logging.getLogger("ddtrace")
+    ddtrace_logger.propagate = False  # prevent ddtrace logger from propagating logs to the root logger
```

Comment:
Reading the [Python Logging docs](https://docs.python.org/3/library/logging.html#logging.Logger.propagate), I think the default is `propagate = True` currently. 

The `ddtrace` logger's parent logger is `root`. I can't think of a situation where this needs to be `True` since this may lead to some unexpected logging behavior when interacting with custom loggers in an application. 

I did some digging around past issues and only found recommendations to disable the propagation, like this: https://github.com/DataDog/dd-trace-py/issues/3306 and https://github.com/DataDog/dd-trace-py/issues/3288.

I'm open to adding an environment variable for `True` if there's a good use case for it though!

Reply:
I don't think we want this to be configurable from an env variable or anything.

I think the better approach is to lock down our libraries logging, and if someone wants to get our logs then they can do the "normal" Python logging library flows, which are:

1. Configure a handler for the "ddtrace" logger
2. Set the "ddtrace" logger to propagate to get handled by the root logger

Reply:
If someone attempts to configure `ddtrace`'s logger before `ddtrace` is imported then would we overwrite this setting? Might result in some confusion for the user. (Not sure if we can detect if a logger has been configured or not)

Also, should we be setting this at all? This overrides the default behaviour of loggers which might also be confusing for a user who attempts to configure their own logging in addition to using `ddtrace`'s log file.

For whichever we decide, we should have a test case to ensure the propagation behaviour.

Reply:
🤔 this is a good point....

I suppose the reason here is to ensure we don't pollute the user's logs or double log, but maybe that is ok?